internal/am/delete_transfer.go

```diff
@@ -20,19 +20,21 @@ type DeleteTransferActivity struct {
 	logger logr.Logger
 }
 
+type DeleteTransferActivityResult struct{}
+
 func NewDeleteTransferActivity(logger logr.Logger, client sftp.Client) *DeleteTransferActivity {
 	return &DeleteTransferActivity{client: client, logger: logger}
 }
 
-func (a *DeleteTransferActivity) Execute(ctx context.Context, params *DeleteTransferActivityParams) error {
+func (a *DeleteTransferActivity) Execute(ctx context.Context, params *DeleteTransferActivityParams) (*DeleteTransferActivityResult, error) {
 	a.logger.V(1).Info("Execute DeleteTransferActivity",
 		"destination", params.Destination,
 	)
 
 	err := a.client.Delete(ctx, params.Destination)
 	if err != nil {
-		return fmt.Errorf("delete transfer: path: %q: %v", params.Destination, err)
+		return nil, fmt.Errorf("delete transfer: path: %q: %v", params.Destination, err)
 	}
 
-	return nil
+	return nil, nil
```

Comment:
I think is fine because we are not using these pointers. When we do actually use the pointer this will no longer be an issue because we would have changed activity result fields  anyway, which means turning this result into a struct. What a nil pointer return signals here is a currently unused value. if that value is used it should panic. I think this is just easier to read as well.

Reply:
What you're saying doesn't seem correct to me. We want to return the struct even if empty because the JSON data converter used by the SDK will encode it as an object (`{}`), which can later have new fields added while maintaining backward compatibility. Returning a nil will result in something like an empty string or similar.

Reply:
I've just tested it, the converter encodes `(*T)(nil)` as an empty string (`""`) whereas `&T{}` becomes: `"{}"`. See how the background-checks app avoids returning `nil`, e.g. see [`SendDeclineEmail`](SendDeclineEmail). Let me know if this makes sense. It's hard to imagine scenarios where this can be relevant (backward and forward compatibility in a distributed environment, rolling upgrades while having running workflow instances, etc), but it seems to be a good idea as a precaution?

```go
package main

import (
	"fmt"
	"log"

	"go.temporal.io/sdk/converter"
)

type Result struct{}

func main() {
	conv := converter.GetDefaultDataConverter()

	r1 := (*Result)(nil)
	if payload, err := conv.ToPayload(r1); err != nil {
		log.Fatal(err)
	} else {
		fmt.Printf("r1: JSON %+q\n", payload.Data)
		// r1: JSON ""
	}

	r2 := &Result{}
	if payload, err := conv.ToPayload(r2); err != nil {
		log.Fatal(err)
	} else {
		fmt.Printf("r2: JSON %+q\n", payload.Data)
		// r2: JSON "{}"
	}
}
```

Reply:
Here's a hypothetical scenario: we deploy an updated version of a workflow function. In our production environment, there are active workflow instances that, upon replay by the workflow worker to restore the state, encounter a situation where they need to deserialize data previously encoded as an empty string (resulting from returning `(*T)(nil)` in older versions of an activity) into a `*T` struct. This is where we would see the discrepancy between the expected data formats: the data converter, tasked with unmarshalling the empty string into a `*T` would fail (panic?).

Reply:
Just an update on this, a few things that I noticed for future prs.

1. We are not using structs in three places, (they are only in local activities so this might be fine anyway) we are using uints
2. This might not be a good solution but alternative I explored is changing the default data converter to a custom data converter that deserializes structs and empty pointers in the same way. It might be useful to explore doing this for a couple reasons and bad for others: 
     - good if we need to switch away from using json at some point (a common use case demonstrated in the temporal docs)
     - bad because it might result in unexpected behavior
 Here are the docs on data conversion: https://docs.temporal.io/dataconversion
 
 Temporal's ToPayload is kind of simple in that it checks if something is an empty interface and if it is it just creates a new payload with no data. We could override this so that we send the zero value of the referent instead just like temporal does with &SomeEmptyActivityResult{}

Reply:
3. Temporal recommends pointers to struct for the scenario's @sevein outlines, that is it means you do not have breaking changes and we can deploy upgraded versions while the services is up. If that is not a use case that we have (which we do not right now) then I am not sure how helpful this PR is.

Reply:
4. If this is an issue, perhaps we should also change workflow definitions to pass empty structs as they currently only return an error. May also be the case for child workflows we design in the future as well.

Reply:
Make sure you bring this up to the team! I checked `preprocessing-sfa` and its child workflow seems to be returning a struct.